src/geff/interops/ctc.py

```diff
@@ -0,0 +1,185 @@
+import shutil
+from pathlib import Path
+
+import numpy as np
+import tifffile
+import typer
+import zarr
+from skimage.measure import regionprops
+from zarr.storage import StoreLike
+
+import geff
+from geff.metadata_schema import GeffMetadata
+from geff.writer_helper import write_props
+
+
+def from_ctc_to_geff(
+    ctc_path: Path,
+    geff_path: Path,
+    segmentation_store: StoreLike | None = None,
+    tczyx: bool = False,
+    overwrite: bool = False,
+) -> None:
+    """
+    Convert a CTC file to a GEFF file.
+
+    Args:
+        ctc_path: The path to the CTC file.
+        geff_path: The path to the GEFF file.
+        segmentation_store: The path or store to the segmentation file.
+                            If not provided, it won't be exported.
+        tczyx: Expand data to make it (T, C, Z, Y, X) otherwise it's (T,) + Frame shape.
+        overwrite: Whether to overwrite the GEFF file if it already exists.
+    """
+    ctc_path = Path(ctc_path)
+    geff_path = Path(geff_path)
+
+    if not ctc_path.exists():
+        raise FileNotFoundError(f"CTC file {ctc_path} does not exist")
+
+    if geff_path.exists() and not overwrite:
+        raise FileExistsError(f"GEFF file {geff_path} already exists")
+
+    if geff_path.exists() and overwrite:
+        shutil.rmtree(geff_path)
+
+    tracks_file_found = False
+
+    for tracks_file in ["man_track.txt", "res_track.txt"]:
+        tracks_file_path = ctc_path / tracks_file
+        if tracks_file_path.exists():
+            tracks_file_found = True
+            break
+
+    if not tracks_file_found:
+        raise FileNotFoundError(
+            f"Tracks file {ctc_path}/man_track.txt or {ctc_path}/res_track.txt does not exist"
+        )
+
+    tracks: dict[int, list[int]] = {}
+    nodes = []
+    edges = []
+    node_dict = {}
+
+    segm_array = None
+    node_id = 0
+
+    sorted_files = sorted(ctc_path.glob("*.tif"))
+    n_1_padding: tuple[int, ...] = ()
+    expand_dims: tuple[None, ...] | slice = slice(None)
+
+    for t, filepath in enumerate(sorted_files):
+        frame = tifffile.imread(filepath)
+
+        if segmentation_store is not None and segm_array is None:
```

Comment:
We should probably also have an option to save the images

Reply:
My only concern is to get the image's path.
I think finding it automatically could be problematic because there are multiple paths the CTC tracking data could be, depending on whether it's ground truth or predictions.